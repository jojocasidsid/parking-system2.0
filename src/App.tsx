import React from 'react'

// packages
import { QueryClient, QueryClientProvider } from 'react-query'
import { SnackbarProvider } from 'notistack'
import { ReactQueryDevtools } from 'react-query/devtools'
import { ThemeProvider } from '@mui/material/styles'

// config
import theme from 'config/theme'
import RouteList from 'config/RouteList'

const queryClient = new QueryClient()
const nodeEnv = process.env.NODE_ENV

const App = () => (
	<QueryClientProvider client={queryClient}>
		<ThemeProvider theme={theme}>
			<SnackbarProvider
				maxSnack={5}
				anchorOrigin={{
					vertical: 'bottom',
					horizontal: 'right',
				}}
				preventDuplicate
				autoHideDuration={5000}
			>
				<RouteList />
				{nodeEnv === 'development' ? (
					<ReactQueryDevtools initialIsOpen={false} />
				) : null}
			</SnackbarProvider>
		</ThemeProvider>
	</QueryClientProvider>
)

export default App
