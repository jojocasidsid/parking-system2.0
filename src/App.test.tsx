import React from 'react'
import renderer from 'react-test-renderer'

import App from './App'

describe('App', () => {
	it('should render app without crashing', () => {
		const tree = renderer.create(<App />)

		expect(tree).toMatchSnapshot()
	})
})
