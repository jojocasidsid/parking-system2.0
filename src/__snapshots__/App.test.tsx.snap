// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app without crashing 1`] = `null`;
